controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -436,3 +437,21 @@ func (r *RKE2ControlPlane) GetConditions() clusterv1.Conditions {
 func (r *RKE2ControlPlane) SetConditions(conditions clusterv1.Conditions) {
 	r.Status.Conditions = conditions
 }
+
+// GetDesiredVersion returns the desired version of the RKE2ControlPlane using Spec.Version field as a default field.
+func (r *RKE2ControlPlane) GetDesiredVersion() string {
+	if r.Spec.Version != "" && isRKE2Version(r.Spec.Version) {
```

Comment:
This does not need validation `isRKE2Version` now that it is ensured on CRD level.